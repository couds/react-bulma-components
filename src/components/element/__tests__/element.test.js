import PropTypes from 'prop-types';
import React from 'react';
import { render } from 'react-testing-library';
import { Element } from '..';

describe('Element component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <Element textColor="white" pull="left">
        Facebook
      </Element>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should accept a react Element as renderAs prop', () => {
    const Custom = props => (
      <p {...props}>
        Custom
        {props.children}
      </p>
    );

    Custom.propTypes = { children: PropTypes.node.isRequired };

    const { asFragment } = render(<Element renderAs={Custom}>This should be a p element</Element>);
    expect(asFragment()).toMatchSnapshot();
  });
});
