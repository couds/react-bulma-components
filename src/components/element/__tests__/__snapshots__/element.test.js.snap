// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element component should accept a react Element as renderAs prop 1`] = `
<DocumentFragment>
  <p>
    CustomThis should be a p element
  </p>
</DocumentFragment>
`;

exports[`Element component should render 1`] = `
<DocumentFragment>
  <div
    class="is-pulled-left has-text-white"
  >
    Facebook
  </div>
</DocumentFragment>
`;
